ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -333,16 +333,19 @@ short getVersion() {
             logger.trace("IndexSegment {} found in bloom filter for index with start offset {} and for key {} ",
                 indexFile.getAbsolutePath(), startOffset, keyToFind);
           }
-          // binary search on the mapped file
-          if (!(serEntries instanceof MappedByteBuffer) || ((MappedByteBuffer) serEntries).isLoaded()) {
-            // isLoaded() will be true only if the entire buffer is in memory - so it being false does not necessarily
-            // mean that the pages in the scope of the search need to be loaded from disk.
-            // Secondly, even if it returned true (or false), by the time the actual lookup is done,
-            // the situation may be different.
-            metrics.mappedSegmentIsLoadedDuringFindCount.inc();
-          } else {
-            metrics.mappedSegmentIsNotLoadedDuringFindCount.inc();
+          if (config.storeIndexMemState == IndexMemState.MMAP_WITH_FORCE_LOAD
+              || config.storeIndexMemState == IndexMemState.MMAP_WITHOUT_FORCE_LOAD) {
+            if (((MappedByteBuffer) serEntries).isLoaded()) {
+              // isLoaded() will be true only if the entire buffer is in memory - so it being false does not necessarily
+              // mean that the pages in the scope of the search need to be loaded from disk.
```

Comment:
we need to remove the comment about isLoaded here since we are no lopnger using isloaded.